src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -71,7 +76,7 @@ struct ExpressionAnalyzerData
   *
   * NOTE: if `ast` is a SELECT query from a table, the structure of this table should not change during the lifetime of ExpressionAnalyzer.
   */
-class ExpressionAnalyzer : protected ExpressionAnalyzerData, private boost::noncopyable
+class ExpressionAnalyzer : public ExpressionAnalyzerData, private boost::noncopyable
```

Comment:
Why so?

Reply:
I think it should have been an all-public immutable struct. "Private data + accessors/modifiers" are for things that change over time and have to maintain complex invariants. On the contrary, the `ExpressionAnalyzerData` is built once at a particular stage of query analysis, and doesn't change. So "all-public immutable" would have been more suitable. But I don't feel like redoing it all now, so for uniformity let's change it back to private add an accessor for window descriptions.